internal/syscallfs/syscallfs_test.go

```diff
@@ -6,15 +6,32 @@ import (
 	"io/fs"
 	"os"
 	"path"
-	"runtime"
+	pathutil "path"
 	"syscall"
 	"testing"
-	"time"
 
-	"github.com/tetratelabs/wazero/internal/platform"
 	"github.com/tetratelabs/wazero/internal/testing/require"
 )
 
+func TestDirFS_Open_Read(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	// Create a subdirectory, so we can test reads outside the FS root.
+	tmpDir = pathutil.Join(tmpDir, t.Name())
+	require.NoError(t, os.Mkdir(tmpDir, 0o700))
+
+	testFS := Adapt(dirFS(tmpDir))
+
+	testFS_Open_Read(t, tmpDir, testFS)
+
+	t.Run("path outside root valid", func(t *testing.T) {
+		_, err := testFS.OpenFile("../foo", os.O_RDONLY, 0)
+
+		// syscall.FS allows relative path lookups
+		require.True(t, errors.Is(err, fs.ErrNotExist))
+	})
+}
```

Comment:
I moved this test here for now so `dirfs_test.go` can be in the `syscallfs_test` package to avoid cyclic dependencies with `syscalltestfs` and `syscallfs`.